src/components/Form/Form.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useState } from 'react';
+import { Todo } from '../../types/Todos';
+import { DefaultErrorsState, User } from '../../types';
+import { TextField } from '../TextField/TextField';
+import { getRandomDigits } from '../../utils';
+import { SelectUser } from '../SelectUser';
+import todos from '../../api/todos';
+
+type Props = {
+  users: User[];
+  addTodo: (newTodo: Todo) => void;
+};
+
+type DefaultState = {
+  titleInput: string;
+  userSelect: string;
+};
+
+const defaultState: DefaultState = {
+  titleInput: '',
+  userSelect: '0',
+};
+
+const defaultErrorState: DefaultErrorsState = {
+  titleInput: '',
+  userSelect: '',
+};
+
+const inputId = `${getRandomDigits()}`;
+
+export const Form: React.FC<Props> = ({ users, addTodo }) => {
+  const [count, setCount] = useState(0);
+  const [state, setState] = useState(defaultState);
+  const [errors, setErrors] = useState(defaultErrorState);
+
+  const maxTodoId = React.useMemo(
+    () => todos.reduce((acc, curr) => Math.max(acc, curr.id), 0),
+    [],
+  );
```

Comment:
The use of `useMemo` here is unnecessary because `todos` is not a changing dependency. You can calculate `maxTodoId` directly without memoization since it doesn't depend on any dynamic state or props.